### PR TITLE
feature: remove TripleRpcException

### PR DESCRIPTION
**Pull request by x-shadow-man**
## What is the purpose of the change
Remove and support dubbo status to triple status mapping for issue #8659 
## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by guohao** [comment]
Plz fix code format

**Comment by x-shadow-man** [comment]
> Plz fix code format

done. pls feel free to review, thx

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java [hunk 1]
```diff
@@ -82,6 +88,31 @@ public static byte toDubboStatus(Code code) {
...
+                code = Code.DEADLINE_EXCEEDED;
+                break;
+            case FORBIDDEN_EXCEPTION:
+                code = Code.PERMISSION_DENIED;
+                break;
+            case LIMIT_EXCEEDED_EXCEPTION:
+                code = Code.ABORTED;
+                break;
+            case TIMEOUT_TERMINATE:
+                code = Code.OUT_OF_RANGE;
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java:104]: DEADLINE_EXCEEDED

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java [hunk 2]
```diff
@@ -82,6 +88,31 @@ public static byte toDubboStatus(Code code) {
...
+        Code code;
+        switch (rpcExceptionCode) {
+            case TIMEOUT_EXCEPTION:
+                code = Code.DEADLINE_EXCEEDED;
+                break;
+            case FORBIDDEN_EXCEPTION:
+                code = Code.PERMISSION_DENIED;
+                break;
+            case LIMIT_EXCEEDED_EXCEPTION:
+                code = Code.ABORTED;
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java:101]: https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md

UNAVAILABLE will be better

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java [hunk 3]
```diff
@@ -82,6 +88,31 @@ public static byte toDubboStatus(Code code) {
...
+            case METHOD_NOT_FOUND:
+                code = Code.NOT_FOUND;
+                break;
+            default:
+                code = Code.UNKNOWN;
+                break;
+        }
+        return fromCode(code);
+    }
+
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java:115]: Add `NETWORK_EXCEPTION` -> `UNAVAILABLE`
`SERIALIZATION_EXCEPTION` -> `INTERNAL`

**Review by guohao — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/UnaryServerStream.java [hunk 4]
```diff
@@ -98,11 +101,11 @@ public void invoke() {
                 try {
                     if (response.hasException()) {
                         final Throwable exception = response.getException();
-                        if (exception instanceof TripleRpcException) {
-                            transportError(((TripleRpcException) exception).getStatus(), response.getObjectAttachments());
+                        if (exception instanceof RpcException) {
+                            transportError(rpcExceptionCodeToGrpc(((RpcException) exception).getCode()), response.getObjectAttachments());
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/UnaryServerStream.java:105]: detail msg in `RPCException` has been dropped, should be pass to `Status`' description

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/UnaryServerStream.java [hunk 5]
```diff
@@ -114,28 +117,28 @@ public void invoke() {
...
                     if (attachments != null) {
                         convertAttachment(trailers, attachments);
                     }
                     getTransportSubscriber().onMetadata(trailers, true);
                 } catch (Throwable e) {
                     LOGGER.warn("Exception processing triple message", e);
-                    if (e instanceof TripleRpcException) {
-                        transportError(((TripleRpcException) e).getStatus(), response.getObjectAttachments());
+                    if (e instanceof RpcException) {
+                        transportError(rpcExceptionCodeToGrpc(((RpcException) e).getCode()), response.getObjectAttachments());
```
Comment (guohao) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/UnaryServerStream.java:137]: Same issue as above

**Comment by EarthChen** [comment]
image

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java [hunk 6]
```diff
@@ -82,6 +83,28 @@ public static byte toDubboStatus(Code code) {
         return status;
     }
 
+    public static GrpcStatus rpcExceptionCodeToGrpc(int rpcExceptionCode) {
```
Comment (EarthChen) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/GrpcStatus.java:94]: Would it be better to replace it with enum？and put them in map and getValue

**Review by EarthChen — CHANGES_REQUESTED** [review]

**Review by EarthChen — APPROVED** [review]
LGTM

**Review by guohao — APPROVED** [review]

**Pull request merged**